File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = {}

        for s in strs:
            d[''.join(sorted(s))] = d.get(''.join(sorted(s)), []) + [s]

            
        return d.values()

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> sorted;

    void dfs(TreeNode* node) {
        if (!node) return;

        dfs(node->left);

        sorted.push_back(node->val);

        dfs(node->right);
    }

    int kthSmallest(TreeNode* root, int k) {
        dfs(root);

        return sorted[k-1];
    }
};
